src/main/java/org/folio/innreach/controller/ExceptionHandlerController.java

```diff
@@ -69,6 +69,12 @@ public Error handleDataIntegrityViolationException(DataIntegrityViolationExcepti
     return createError(HttpStatus.BAD_REQUEST, e.getMessage());
   }
 
+  @ExceptionHandler(IllegalArgumentException.class)
+  @ResponseStatus(HttpStatus.BAD_REQUEST)
+  public Error handleIllegalArgumentException(IllegalArgumentException e) {
```

Comment:
IllegalArgumentException is too general to me mapped to BAD_REQUEST always.
I'd suggest to intercept the exception where it happens and transform to some specific one. As I see from the code of `InnReachExternalServiceImpl` it uses `InnReachException` to signal about invalid data. This specific exception could be what's needed